driver-core/src/test/unit/com/mongodb/connection/InternalStreamConnectionSpecification.groovy

```diff
@@ -471,6 +472,66 @@ class InternalStreamConnectionSpecification extends Specification {
                         new BsonDocument('ok', new BsonInt32(1)), 1000)])
     }
 
+    def 'should extract cluster and operation time into session context'() {
+        given:
+        def connection = getOpenedConnection()
+        def pingCommandDocument = new BsonDocument('ping', new BsonInt32(1))
+        def commandMessage = new SimpleCommandMessage('admin.$cmd', pingCommandDocument, primary(), messageSettings)
+        def response = '''{
+                            ok : 1,
+                            operationTime : { $timestamp : { "t" : 40, "i" : 20 } },
+                            $clusterTime :  { clusterTime : { $timestamp : { "t" : 42, "i" : 21 } } }
+                          }'''
+        stream.getBuffer(1024) >> { new ByteBufNIO(ByteBuffer.wrap(new byte[1024])) }
+        stream.read(16) >> helper.defaultMessageHeader(commandMessage.getId())
+        stream.read(_) >> helper.reply(response)
+        def sessionContext = Mock(SessionContext) {
+            1 * advanceOperationTime(BsonDocument.parse(response).getTimestamp('operationTime'))
+            1 * advanceClusterTime(BsonDocument.parse(response).getDocument('$clusterTime'))
+        }
+
+        when:
+        connection.sendAndReceive(commandMessage, new BsonDocumentCodec(), sessionContext)
+
+        then:
+        true
```

Comment:
Comment: then: true - looks a little odd, we generally put the mock assertions here, but either way works - so no change needed.